### PR TITLE
NEW GRA testcase using new ID: 17

### PR DESCRIPTION
**Pull request by harinineelu**
Also includes default.config

### CODE REVIEW

File: src/harness/testcases/WINNF_FT_S_GRA_testcase.py [hunk 1]
```diff
@@ -997,3 +999,217 @@ def test_WINNF_FT_S_GRA_16(self):
...
+              'longitude'] = getRandomLatLongInPolygon(ppa_record)
+    for device_index in devices_in_ppa_zone:
+      devices[device_index]['installationParam']['latitude'], devices[
+          device_index]['installationParam'][
+              'longitude'] = getRandomLatLongInPolygon(ppa_record)
+
+    config = {
+        'registrationRequest': devices,
+        'conditionalRegistrationData': conditionals,
+        'ppa': {
```
Comment (janetliuboston) [inline src/harness/testcases/WINNF_FT_S_GRA_testcase.py]: For N2 >= 0 PPAs, should PPAs be an array?

File: src/harness/testcases/WINNF_FT_S_GRA_testcase.py [hunk 2]
```diff
@@ -997,3 +999,217 @@ def test_WINNF_FT_S_GRA_16(self):
...
+      self._sas_admin.InjectUserId({'userId': device['userId']})
+
+    # Pre-load conditional registration data for N3 CBSDs.
+    if ('conditionalRegistrationData' in config) and (
+        config['conditionalRegistrationData']):
+      self._sas_admin.PreloadRegistrationData(
+          config['conditionalRegistrationData'])
+
+    # Inject PAL database record
+    for pal_record in config['ppa']['palRecords']:
```
Comment (janetliuboston) [inline src/harness/testcases/WINNF_FT_S_GRA_testcase.py]: There might be 0 PPAs. May want to check for the existence of config['ppa'] before accessing it. Same applies wherever config['ppa'] is referenced below.

File: src/harness/testcases/WINNF_FT_S_GRA_testcase.py [hunk 3]
```diff
@@ -997,3 +999,217 @@ def test_WINNF_FT_S_GRA_16(self):
...
+    # Update PPA record with device's CBSD ID and Inject zone data
+    for device_index in config['ppa']['ppaClusterList']:
+      config['ppa']['ppaRecord']['ppaInfo']['cbsdReferenceId'] = [
+          cbsd_ids[device_index]
+      ]
+    zone_id = self._sas_admin.InjectZoneData({
+        'record': config['ppa']['ppaRecord']
+    })
+    self.assertTrue(zone_id)
+
```
Comment (janetliuboston) [inline src/harness/testcases/WINNF_FT_S_GRA_testcase.py]: Missing trigger CPAS if N2 PPAs >= 0.
  if ('ppa' in config) and (config['ppa']): # and possibly (len(config['ppa']] > 0)
    # Trigger daily activities
    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()

File: src/harness/testcases/WINNF_FT_S_GRA_testcase.py (hunk 1 above)
Comment (harinineelu) [reply src/harness/testcases/WINNF_FT_S_GRA_testcase.py]: Fixed.

File: src/harness/testcases/WINNF_FT_S_GRA_testcase.py (hunk 2 above)
Comment (harinineelu) [reply src/harness/testcases/WINNF_FT_S_GRA_testcase.py]: Good point, Fixed.

File: src/harness/testcases/WINNF_FT_S_GRA_testcase.py (hunk 3 above)
Comment (harinineelu) [reply src/harness/testcases/WINNF_FT_S_GRA_testcase.py]: Fixed.

**Comment by harinineelu** [comment]
Thanks for the careful review, Janet. Fixed all. 

Could you please take a look. Thanks!

**Comment by harinineelu** [comment]
Merged #429

**Pull request merged**